spring-cloud-stream-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc

```diff
@@ -81,147 +71,586 @@ public class StreamApplicationTests {
+  A list of Zookeeper nodes for the Kafka binder to connect to. Default `localhost`.
+spring.cloud.stream.binder.kafka.defaultZkPort::
+  The list of Zookeeper nodes allows to specify hosts with or without port information, i.e. `host1,host2:port2`. This configuration sets the default port when no port is configured in the node list. Default `2181`.
+spring.cloud.stream.binder.kafka.headers::
+  The list of custom that will be transported by the binder. Default empty.
+spring.cloud.stream.binder.kafka.offsetUpdateTimeWindow::
+  The frequency in milliseconds with which offsets are saved. Ignored if `0`. Default `10000`.
+spring.cloud.stream.binder.kafka.offsetUpdateCount::
+  The frequency in number of updates, which which consumed offsets are persisted. Ignored if `0`. Default `0`. Mutually exclusive with `offsetUpdateTimeWindow`.
+spring.cloud.stream.binder.kafka.requiredAcks::
+  The number of required acks on the broker.
+
+==== Kafka Consumer Properties
+
+The following properties are available for Kafka consumers only and
+must be prefixed with `spring.cloud.stream.bindings.<channelName>.`
+
+autoCommitOffset::
+  True to autocommit offsets when a message has been processed. If set to false, an `Acknowledgment` header will be available in the message headers for late acknowledgment. Default `true`.
+mode::
+    When set to `raw`, will disable header parsing on input. Useful when inbound data is coming from outside Spring Cloud Stream applications. Default `embeddedHeaders`.
+resetOffsets::
+  True to reset offsets on the consumer to the value provided by `startOffset`. Default `false`.
+startOffset::
+  The starting offset for new groups or when `resetOffsets` is `true`. Allowed values: `earliest`,`latest`. Defaults to null (equivalent to earliest).
+minPartitionCount::
+  The minimum number of partitions expected by the consumer if it creates the consumed topic automatically. Defaults to `1`.
+
+
+
+==== Kafka Producer Properties
+
+The following properties are available for Kafka producers only and
+must be prefixed with `spring.cloud.stream.bindings.<channelName>.`
+
+bufferSize::
+  This is an upper limit of how much data the Kafka Producer will attempt to batch before sending – specified in bytes. Default `16384`.
+sync::
+  Whether the producer is synchronous. Defaults to `false`.
+batchTimeout::
+  How long will the producer wait before sending in order to allow more messages to get accumulated in the same batch. Normally the producer will not wait at all, and simply send all the messages that accumulated while the previous send was in progress. A non-zero value may increase throughput at the expense of latency. Default `0`.
+mode::
+    When set to `raw`, disable header propagation on output. Useful when producing data for non-Spring Cloud Stream applications. Default `embeddedHeaders`.
+
+== Binder detection
 
 Spring Cloud Stream relies on implementations of the Binder SPI to perform the task of connecting channels to message
```

Comment:
> Binder SPI

It'd be better to bring "Binder SPI" section towards the beginning, so we could then link it here.